src/Atmos/Model/AtmosModel.jl

```diff
@@ -594,6 +596,22 @@ function update_auxiliary_state!(
 
     update_auxiliary_state!(nodal_update_auxiliary_state!, dg, m, Q, t, elems)
 
+    # Filter the highest order vertical mode from density
+    grid = dg.grid
+    if m.ref_state isa HydrostaticState
+        target = AtmosFilterDensityPerturbation(m)
+    else
+        target = (:projected_ρ,)
+    end
+    Filters.apply!(
+        state_auxiliary,
+        target,
+        grid,
+        Filters.CutoffFilter(grid),
```

Comment:
Question for my understanding: where is the cutoff to N-1 specified? I'm used to having to do `CutoffFilter(grid, polynomialorder(grid) - 1)`

Reply:
The docs https://github.com/CliMA/ClimateMachine.jl/blob/2be057f82576c4ec3f95243854acc8662872c4a3/src/Numerics/Mesh/Filters.jl#L214-L219
say that without it will cut off the highest polynomial mode.

Guessing the line you have was my fault, maybe we are being too aggressive in the ocean code(?).

Reply:
@sandreza does it make sense that we're preserving discrete incompressibility if we are actually zeroing out the top two modes instead of just the top mode? because we seem to still be preserving it based on our tests